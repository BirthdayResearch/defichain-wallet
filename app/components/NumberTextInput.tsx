import * as Localization from 'expo-localization'
import React from 'react'
import { KeyboardTypeOptions, Platform, TextInputProps } from 'react-native'
import { useThemeContext } from '../contexts/ThemeProvider'
import { tailwind } from '../tailwind'
import { TextInput } from './index'

export function NumberTextInput (props: React.PropsWithChildren<TextInputProps>): JSX.Element {
  const { isLight } = useThemeContext()
  const { style, ...otherProps } = props
  let keyboardType: KeyboardTypeOptions = 'numeric'
  if (Platform.OS === 'ios' && Localization.decimalSeparator !== '.') {
    keyboardType = 'default'
  }
  return (
    <TextInput
      placeholderTextColor={isLight ? 'rgba(0, 0, 0, 0.4)' : '#828282'}
      style={[style, tailwind(isLight ? 'text-gray-700' : 'text-white')]}
      {...otherProps}
      keyboardType={keyboardType}
    />
  )
}
