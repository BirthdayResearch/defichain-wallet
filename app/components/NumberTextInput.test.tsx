import { render } from "@testing-library/react-native";
import * as Localization from 'expo-localization';
import * as React from "react";
import { Platform } from 'react-native'
import { NumberTextInput } from "./NumberTextInput"

jest.mock("../contexts/ThemeProvider")
jest.mock('expo-localization')

const platform = ['ios', 'android']

describe('NumberTextInput', () => {
  platform.forEach((os) => {

    it(`should match snapshot with OS ${os} period decimal locale`, () => {
      (Platform as any).OS = os;
      (Localization as any).decimalSeparator = '.'
      const component = (
        <NumberTextInput value={'123'} editable={true} placeholder='Enter an amount' />
      )
      const rendered = render(component)
      expect(rendered.toJSON()).toMatchSnapshot()
    })

    it(`should match snapshot with OS ${os} period comma locale`, () => {
      (Platform as any).OS = os;
      (Localization as any).decimalSeparator = ','
      const component = (
        <NumberTextInput value={'123'} editable={true} placeholder='Enter an amount' />
      )
      const rendered = render(component)
      expect(rendered.toJSON()).toMatchSnapshot()
    })

    it(`should match snapshot with OS ${os} period unknown locale`, () => {
      (Platform as any).OS = os;
      (Localization as any).decimalSeparator = ' '
      const component = (
        <NumberTextInput value={'123'} editable={true} placeholder='Enter an amount' />
      )
      const rendered = render(component)
      expect(rendered.toJSON()).toMatchSnapshot()
    })
  })

})
