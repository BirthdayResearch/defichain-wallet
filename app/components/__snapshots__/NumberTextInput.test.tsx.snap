// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberTextInput should match snapshot with OS android period comma locale 1`] = `
<TextInput
  allowFontScaling={true}
  editable={true}
  keyboardType="numeric"
  placeholder="Enter an amount"
  placeholderTextColor="rgba(0, 0, 0, 0.4)"
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "fontFamily": "RegularFont",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      },
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
  value="123"
/>
`;

exports[`NumberTextInput should match snapshot with OS android period decimal locale 1`] = `
<TextInput
  allowFontScaling={true}
  editable={true}
  keyboardType="numeric"
  placeholder="Enter an amount"
  placeholderTextColor="rgba(0, 0, 0, 0.4)"
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "fontFamily": "RegularFont",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      },
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
  value="123"
/>
`;

exports[`NumberTextInput should match snapshot with OS android period unknown locale 1`] = `
<TextInput
  allowFontScaling={true}
  editable={true}
  keyboardType="numeric"
  placeholder="Enter an amount"
  placeholderTextColor="rgba(0, 0, 0, 0.4)"
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "fontFamily": "RegularFont",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      },
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
  value="123"
/>
`;

exports[`NumberTextInput should match snapshot with OS ios period comma locale 1`] = `
<TextInput
  allowFontScaling={true}
  editable={true}
  keyboardType="default"
  placeholder="Enter an amount"
  placeholderTextColor="rgba(0, 0, 0, 0.4)"
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "fontFamily": "RegularFont",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      },
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
  value="123"
/>
`;

exports[`NumberTextInput should match snapshot with OS ios period decimal locale 1`] = `
<TextInput
  allowFontScaling={true}
  editable={true}
  keyboardType="numeric"
  placeholder="Enter an amount"
  placeholderTextColor="rgba(0, 0, 0, 0.4)"
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "fontFamily": "RegularFont",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      },
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
  value="123"
/>
`;

exports[`NumberTextInput should match snapshot with OS ios period unknown locale 1`] = `
<TextInput
  allowFontScaling={true}
  editable={true}
  keyboardType="default"
  placeholder="Enter an amount"
  placeholderTextColor="rgba(0, 0, 0, 0.4)"
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "fontFamily": "RegularFont",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      },
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
  value="123"
/>
`;
